### PR TITLE
Setup devcontainer

### PR DESCRIPTION
**Pull request by dorianvp**
This PR adds a devcontainer to simplify project setup and contributions.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+RUN node -v
+RUN npm -v
+RUN corepack enable
+RUN yarn --version
+
+# Stage 2: Install Rust and Nextest
+FROM docker.io/rust:latest as rust-stage
+
+# Install cargo-nextest
+RUN cargo install cargo-nextest
```
Comment (juanky201271) [inline .devcontainer/Dockerfile]: I got this: `Error: there is no registered task type 'cargo'. Did you miss installing an extension that provides a corresponding task provider?`

Comment (zancas) [reply .devcontainer/Dockerfile]: Huh...   what's going on with this @dorianvp ?  Do you also get that error?  Given the base image, I am surprised at the error.   I would assume `cargo` is installed.   Does this happen every time @juanky201271 ?  Do you get the same error @Edicksonjga ?

**Comment by zancas** [comment]
Any motion on this?

**Comment by dorianvp** [comment]
I think the best way forward is to break this PR into two:

1. One for the `devcontainer.json` config file for VS Code.
2. Another for the CLI runners.

This will keep things cleaner.

**Review by Edicksonjga — APPROVED** [review]

**Pull request merged**